Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift

```diff
@@ -75,26 +77,73 @@ public extension CVPixelBuffer {
+
+        var _copy: CVPixelBuffer?
+
+        let width = CVPixelBufferGetWidth(self)
+        let height = CVPixelBufferGetHeight(self)
+        let formatType = CVPixelBufferGetPixelFormatType(self)
+        let attachments = CVBufferGetAttachments(self, .shouldPropagate)
+
+        CVPixelBufferCreate(nil, width, height, formatType, attachments, &_copy)
+
+        guard let copy = _copy else {
+            throw Error.allocationFailed
+        }
+
+        CVPixelBufferLockBaseAddress(self, .readOnly)
+        CVPixelBufferLockBaseAddress(copy, [])
+
         defer {
-            copy.unlockBaseAddress(options: [])
-            self.unlockBaseAddress(options: .readOnly)
+            CVPixelBufferUnlockBaseAddress(copy, [])
+            CVPixelBufferUnlockBaseAddress(self, .readOnly)
         }
-        
-        // TODO: Handle non-planar pixel buffers
-        for plane in 0 ..< self.planeCount {
-            let dest = copy.baseAddress(of: plane)
-            let source = self.baseAddress(of: plane)
-            let height = self.height(of: plane)
-            let bytesPerRow = self.bytesPerRow(of: plane)
-            memcpy(dest, source, height * bytesPerRow)
+
+        let pixelBufferPlaneCount: Int = CVPixelBufferGetPlaneCount(self)
+
+
+        if pixelBufferPlaneCount == 0 {
+            let dest = CVPixelBufferGetBaseAddress(copy)
+            let source = CVPixelBufferGetBaseAddress(self)
+            let height = CVPixelBufferGetHeight(self)
+            let bytesPerRowSrc = CVPixelBufferGetBytesPerRow(self)
+            let bytesPerRowDest = CVPixelBufferGetBytesPerRow(copy)
+            if bytesPerRowSrc == bytesPerRowDest {
+                memcpy(dest, source, height * bytesPerRowSrc)
+            } else {
+                var startOfRowSrc = source
+                var startOfRowDest = dest
+                for _ in 0..<height {
+                    memcpy(startOfRowDest, startOfRowSrc, min(bytesPerRowSrc, bytesPerRowDest))
+                    startOfRowSrc = startOfRowSrc?.advanced(by: bytesPerRowSrc)
+                    startOfRowDest = startOfRowDest?.advanced(by: bytesPerRowDest)
+                }
+            }
+
+        } else {
+            for plane in 0 ..< pixelBufferPlaneCount {
+                let dest        = CVPixelBufferGetBaseAddressOfPlane(copy, plane)
+                let source      = CVPixelBufferGetBaseAddressOfPlane(self, plane)
+                let height      = CVPixelBufferGetHeightOfPlane(self, plane)
+                let bytesPerRowSrc = CVPixelBufferGetBytesPerRowOfPlane(self, plane)
+                let bytesPerRowDest = CVPixelBufferGetBytesPerRowOfPlane(copy, plane)
+
+                if bytesPerRowSrc == bytesPerRowDest {
+                    memcpy(dest, source, height * bytesPerRowSrc)
+                } else {
+                    var startOfRowSrc = source
+                    var startOfRowDest = dest
+                    for _ in 0..<height {
```

Comment:
```suggestion
                    for _ in 0 ..< height {
```

Reply:
it's inconsistent with [this](https://github.com/s1ddok/SwiftyCVPixelBuffer/blob/4f2f0deb8cd5274692010e6c2f9a183e0d4f9e69/Sources/SwiftyCVPixelBuffer/CVPixelBuffer%2BUtils.swift#L90)